pkg/encoding/v1/encoding.go

```diff
@@ -98,9 +100,19 @@ func (pt *PortType) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type Uri string
+
+// TODO: Add Uri unmarshalling to validate it
+
+type UriPath string
```

Comment:
@surajssd @kadel you are right, calling it even URN is to broad. Host should be `Fqdn` (as defined by RFC 3986) and path can also be a regexp so it should be `PathRegex` (as defined by IEEE Std 1003.1)